lib/Backend/GlobOptBlockData.cpp

```diff
@@ -144,6 +144,10 @@ GlobOptBlockData::ReuseBlockData(GlobOptBlockData *fromData)
 
     this->changedSyms = fromData->changedSyms;
     this->capturedValues = fromData->capturedValues;
+    if (this->capturedValues)
+    {
+        this->capturedValues->IncrementRefCount();
```

Comment:
>IncrementRefCount [](start = 30, length = 17)

We won't call delete on the previous block, so I don't think you should inc here.

Reply:
I see, ok.

Reply:
I'm thinking decrementing refCount in `NullOutBlockData` (we will call `NullOutBlockData` on the previous block instead of, as you said, 'DeleteBlockData`). That will keep the refCount accurate and still enable deletion of capturedValues. What do you think?